### PR TITLE
Nit fixes, cleanup old versions

### PR DESCRIPTION
**Pull request by jwhitlock**
This fixes a few nits and warnings I noticed when running locally:

* I'm ignoring `dump.rdb`, created when I run `redis-server` in the directory so tests can access it.
* The documentation build complained about `language = None`. The default of `en` works for us. Also bumped the copyright.
* Added `Framework :: Django :: 4.2` to the classifiers
* Django 3.2 doesn't claim support for Python 3.11. Tests pass, but there are some deprecation warnings. I removed it from the matrix.
* I removed the dropped Flask and Sanic versions from the commands
* I moved the Django-specific pytest options from `pytest.ini` to `tox.ini`, to remove some warnings from the Flask and Sanic builds about unknown options.

There's more I _want_ to do (convert from `setup.py` to `pyproject.toml`, figure out why a tox run opens so many files), but I'm stopping myself.

All these are nits, so I'm willing to drop any controversial ones.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,3 +10,4 @@ build
 dist
 *.pyc
 .pytest_cache/
+dump.rdb
```
Comment (grahamalama) [inline .gitignore:13]: Do we know what this was about?

File: docs/conf.py [hunk 2]
```diff
@@ -60,7 +60,7 @@
 
 # General information about the project.
 project = "python-dockerflow"
-copyright = "2016, Mozilla Foundation"
+copyright = "2016 - 2023, Mozilla Foundation"
```
Comment (grahamalama) [inline docs/conf.py:63]: Should we calculate this end year?

File: docs/conf.py [hunk 3]
```diff
@@ -77,7 +77,7 @@
 #
 # This is also used if you do content translation via gettext catalogs.
 # Usually you set "language" from the command line for these cases.
-language = None
+# language = "en"
```
Comment (grahamalama) [inline docs/conf.py:80]: Did we mean to comment this line out? Asking because we also changed the value to `"en"`.

File: setup.py [hunk 4]
```diff
@@ -30,6 +30,7 @@ def read(*parts):
         "Framework :: Django :: 3.2",
         "Framework :: Django :: 4.0",
         "Framework :: Django :: 4.1",
+        "Framework :: Django :: 4.2",
```
Comment (grahamalama) [inline setup.py:33]: Oops!

**Review by grahamalama — DISMISSED** [review]

File: .gitignore (hunk 1 above)
Comment (jwhitlock) [reply .gitignore:13]: This is the file created by running `redis-server` in the same folder as the project.

File: docs/conf.py (hunk 3 above)
Comment (jwhitlock) [reply docs/conf.py:80]: I agree, it is confusing. "en" is the default, and also appropriate for our project. I'll uncomment.

**Comment by jwhitlock** [comment]
I've answered some questions, fixed some issues, and also added a `.readthedocs.yaml` to prepare for this file being required in September: https://blog.readthedocs.com/migrate-configuration-v2/#migrate-your-project-to-readthedocs-yaml-configuration-file-v2

**Pull request merged**